src/frequenz/client/common/pagination/pagination.py

```diff
@@ -0,0 +1,100 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the pagination used with the common client."""
+
+from __future__ import annotations  # required for constructor type hinting
+
+import logging
+from dataclasses import dataclass
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_info_pb2 import (
+    PaginationInfo as PBPaginationInfo,
+)
+
+# pylint: disable=import-error, no-name-in-module
+from frequenz.api.common.v1.pagination_params_pb2 import (
+    PaginationParams as PBPaginationParams,
+)
+
+# Set up logging
+_logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PaginationParams:
```

Comment:
I think here we need to make another design decision. Since in Python we already have the module name (`pagination`) maybe it is too redundant to make the class `PaginationParams` and we can just use `Params` to encourage the google-style imports `from frequenz.client.common import pagination; p = pagination.Params(...)`.

I think because of how proto files are structured, this makes sense.

FYI @frequenz-floss/python-sdk-team